tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts

```diff
@@ -183,3 +207,86 @@ export function conv2dWithIm2Col({
 
   return out;
 }
+
+export function conv2DImpl({
+  x,
+  filter,
+  convInfo,
+  backend,
+  bias = null,
+  preluActivationWeights = null,
+  leakyreluAlpha = 0,
+  activation = null
+}: Conv2DConfig) {
+  const hasBias = bias != null;
+  const hasPreluActivationWeights = preluActivationWeights != null;
+
+  let program: Conv2DMMProgram|Conv2DNaiveProgram|Conv2DMMVec4Program;
+  const sameSize = convInfo.filterHeight === convInfo.inHeight &&
+                   convInfo.filterWidth === convInfo.inWidth &&
+                   convInfo.padInfo.type === 'VALID';
+  if (sameSize || (convInfo.filterHeight === 1 && convInfo.filterWidth === 1 &&
+      convInfo.dilationHeight === 1 && convInfo.dilationWidth === 1 &&
+      convInfo.strideHeight === 1 && convInfo.strideWidth === 1 &&
+      (convInfo.padInfo.type === 'SAME' || convInfo.padInfo.type === 'VALID')))
+  {
+    return conv2dByMatMul({
+      x,
+      filter,
+      convInfo,
+      backend,
+      bias,
+      activation,
+      preluActivationWeights,
+      leakyreluAlpha
+    });
+  }
+
+  if (env().getBool('WEBGPU_CONV_SEPARATE_IM2COL_SHADER') && x.shape[0] === 1) {
```

Comment:
In original code, this flag is only used by Conv2D. Could fusedConv2D also use conv2dWithIm2Col here?

Reply:
Yes, it also works for `fusedConv2d`.